paper/paper.Rmd

```diff
@@ -0,0 +1,221 @@
+---
+title: "Static Code Analysis for R"
+date: "`r Sys.Date()`"
+tags: ["R", "linter", "tidyverse"]
+authors:
+  - name: Jim Hester
+    affiliation: 1
+    orcid: ~
+  - name: Michael Chirico
+    affiliation: 2
+    orcid: 0000-0003-0787-087X
+  - name: Alexander Rosenstock
+    affiliation: 3
+    orcid: ~
+  - name: Indrajeet Patil
+    orcid: 0000-0003-1995-6531
+    affiliation: 4
+affiliations:
+  - index: 1
+    name: Netflix
+  - index: 2
+    name: Google
+  - index: 3
+    name: Mathematisches Institut der Heinrich-Heine-Universität Düsseldorf
+  - index: 4
+    name: Preisenergie GmbH, Munich, Germany
+output: 
+    md_document:
+      variant: "markdown"
+      preserve_yaml: true
+      standalone: true
+bibliography: paper.bib
+csl: apa.csl
+link-citations: yes
+---
+
+```{r setup, warning=FALSE, message=FALSE, echo=FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  out.width = "100%",
+  comment = "#>"
+)
+
+library(lintr)
+
+withr::local_options(list(
+  lintr.format_width = 60L
+))
+```
+
+# Statement of Need
+
+R is an interpreted, dynamically-typed programming language [@base2023]. It is a popular choice for statistical analysis and visualization, and is used by a wide range of researchers and data scientists. The `{lintr}` package is an open-source R package that provides static code analysis to check for a variety of common problems related to readability, efficiency, consistency, style, etc. In particular, by default it enforces the tidyverse style guide [@Wickham2023]. It is designed to be easy to use and integrate into existing workflows, and can be used as part of an automated build or continuous integration process. `{lintr}` also integrates with a number of popular IDEs and text editors, such as RStudio and Visual Studio Code, making it convenient for users to run `{lintr}` checks on their code as they work.
```

Comment:
Should we add a citation about "static code analysis", e.g. just the Wikipedia page?

Reply:
Similarly, does it make sense to mention popular linters in other languages? I'm not sure about how general the context we should provide for JOSS is.

Reply:
There is a comprehensive summary of tools here: https://github.com/analysis-tools-dev/static-analysis

But I can't think of a good place where to place it, or how to cite it.